src/Accounts/Accounts/CommonModule/ContextAdapter.cs

```diff
@@ -195,30 +195,32 @@ public object GetParameterValue(string resourceId, string moduleName, Invocation
         internal async Task<HttpResponseMessage> AuthenticationHelper(IAzureContext context, string endpointResourceIdKey, string endpointSuffixKey, HttpRequestMessage request, CancellationToken cancelToken, Action cancelAction, SignalDelegate signal, NextDelegate next, TokenAudienceConverterDelegate tokenAudienceConverter = null)
         {
             IAccessToken accessToken = await AuthorizeRequest(context, request, cancelToken, endpointResourceIdKey, endpointSuffixKey, tokenAudienceConverter);
-            var newRequest = await request.CloneWithContentAndDispose(request.RequestUri, request.Method);
-            var response = await next(request, cancelToken, cancelAction, signal);
-
-            if (response.MatchClaimsChallengePattern())
+            using (var newRequest = await request.CloneWithContent(request.RequestUri, request.Method))
             {
-                //get token again with claims challenge
-                if (accessToken is IClaimsChallengeProcessor processor)
+                var response = await next(request, cancelToken, cancelAction, signal);
+
+                if (response.MatchClaimsChallengePattern())
                 {
-                    try
+                    //get token again with claims challenge
+                    if (accessToken is IClaimsChallengeProcessor processor)
                     {
-                        var claimsChallenge = ClaimsChallengeUtilities.GetClaimsChallenge(response);
-                        if (!string.IsNullOrEmpty(claimsChallenge))
+                        try
                         {
-                            await processor.OnClaimsChallenageAsync(newRequest, claimsChallenge, cancelToken).ConfigureAwait(false);
-                            response = await next(newRequest, cancelToken, cancelAction, signal);
+                            var claimsChallenge = ClaimsChallengeUtilities.GetClaimsChallenge(response);
+                            if (!string.IsNullOrEmpty(claimsChallenge))
+                            {
+                                await processor.OnClaimsChallenageAsync(newRequest, claimsChallenge, cancelToken).ConfigureAwait(false);
+                                response = await next(newRequest, cancelToken, cancelAction, signal);
```

Comment:
I double check the code and think we need to dispose the old response and request here.